be/src/vec/functions/array/function_array_exists.cpp

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <vec/columns/column_array.h>
+#include <vec/columns/column_nullable.h>
+#include <vec/columns/columns_number.h>
+#include <vec/data_types/data_type_array.h>
+#include <vec/data_types/data_type_number.h>
+#include <vec/functions/function.h>
+#include <vec/functions/function_helpers.h>
+#include <vec/functions/simple_function_factory.h>
+
+#include <vec/utils/util.hpp>
+
+namespace doris::vectorized {
+
+// array_exists([1, 2, 3, 0]) -> [1, 1, 1, 0]
+class FunctionArrayExists : public IFunction {
+public:
+    static constexpr auto name = "array_exists";
+
+    static FunctionPtr create() { return std::make_shared<FunctionArrayExists>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        DCHECK(is_array(arguments[0]))
```

Comment:
The return type of 'array_exists' is ARRAY_BOOLEAN which defined in 'doris_builtins_functions.py', why does return 'arguments[0]' ?

Reply:
> The return type of 'array_exists' is ARRAY_BOOLEAN which defined in 'doris_builtins_functions.py', why does return 'arguments[0]' ?

Because array_exists input  params is also ARRAY_BOOLEAN.  In fe, array_map result  is casted to ARRAY_BOOLEAN